app/hero.py

```diff
@@ -0,0 +1,85 @@
+from __future__ import annotations
+
+
+class Hero:
+    def __init__(
```

Comment:
it is overall not a good concept to have one superclass that does it all.
Your code should be more modular - create separate classes for equipment to offload some logic from here and make it more reusable